pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees latitude.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.Sphere
+
+    Examples
+    --------
+    Create a solid sphere.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphere()
+    >>> solid_sphere.plot(show_edges=True)
+
+    A solid sphere is 3D in comparison to the 2d `pyvista.Sphere`.
+    Generate a solid hemi-sphere to see the internal structure.
+
+    >>> isinstance(solid_sphere, pv.UnstructuredGrid)
+    True
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180, end_theta=360
+    ... )
+    >>> partial_solid_sphere.plot(show_edges=True)
+
+    To see the cell structure inside the solid sphere,
+    only 1/4 of the sphere is generated. The cells exploded
+    and colored by radial position.
+
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180,
+    ...     end_theta=360,
+    ...     start_phi=0,
+    ...     end_phi=90,
+    ...     radius_resolution=5,
+    ...     theta_resolution=8,
+    ...     phi_resolution=8,
+    ... )
+    >>> partial_solid_sphere["cell_radial_pos"] = np.linalg.norm(
+    ...     partial_solid_sphere.cell_centers().points, axis=-1
+    ... )
+    >>> partial_solid_sphere.explode(1).plot()
+
+    """
+    if radius is None:
+        radius = np.linspace(inner_radius, outer_radius, radius_resolution)
+    if theta is None:
+        theta = np.linspace(start_theta, end_theta, theta_resolution)
+    if phi is None:
+        phi = np.linspace(start_phi, end_phi, phi_resolution)
+
+    if radius[0] < 0:
+        raise ValueError("minimum radius cannot be negative")
+    if theta[0] < 0:
+        raise ValueError("minimum theta cannot be negative")
+    if phi[0] < 0:
+        raise ValueError("minimum phi cannot be negative")
+    if theta[-1] > 360:
+        raise ValueError("maximum theta cannot be > 360")
+    if phi[-1] > 180:
+        raise ValueError("maximum phi cannot be > 180")
+
+    # Hereafter all degrees are in radians
+    theta = np.deg2rad(theta)
+    phi = np.deg2rad(phi)
+
+    nr = len(radius)
+    ntheta = len(theta)
+    nphi = len(phi)
+
+    if nr < 2:
+        raise ValueError("radius_resolution must be 2 or more")
+    if ntheta < 2:
+        raise ValueError("theta_resolution must be 2 or more")
+    if nphi < 2:
+        raise ValueError("phi_resolution must be 2 or more")
+
+    def _spherical_to_cartesian(r, phi, theta):
+        """Convert spherical coordinate sequences to a ``(n,3)`` cartesian coordinate array.
```

Comment:
PyVista is a project to wrapper library for vtk, it would be good to prepare [vtkSphericalTransform](https://vtk.org/doc/nightly/html/classvtkSphericalTransform.html) wrapper in another PR and replace this.

Reply:
I think this is a case where vtk is not preferred.  vtkSphericalTransform takes in one point at a time.  This will be slow and awkward when we have to iterate over this in Python.

I will admit to my laziness in that we have this function 
https://github.com/pyvista/pyvista/blob/18c921e083f4611a0f1e330ec7a479532934545f/pyvista/core/utilities/features.py#L174-L175

But I could not understand it with the flipping of (z, y, x) and the meaning of u, v, w.  It sounds like this is heavily geared to geoscience use, which my brain doesn't natively understand.    

It also doesn't seem consistent with the inverse function.

https://github.com/pyvista/pyvista/blob/18c921e083f4611a0f1e330ec7a479532934545f/pyvista/core/utilities/features.py#L212

I agree on the point that there is a need for a comprehensive functionality for this operation.

Reply:
Thank you. Let's consider this discussion in #4979. This PR is fine AS IS.